Systematize multiline strings for message text in hl_api

stinebuu opened the pull request:
This PR fixes #1178.

Systematize multiline strings for message text in hl_api_*. Have also cleaned up some texts to make them easier to read when looking at the code.

clinssen commented on pynest/nest/lib/hl_api_helper.py:
```diff
@@ -514,9 +513,8 @@ def model_deprecation_warning(model):
-            future version of NEST, use {1} instead.\
-            ".format(model, _deprecation_warning[model]['replacement'])
+            text = """The {0} model is deprecated and will be removed in a future version of NEST, use {1} instead.
+            """.format(model, _deprecation_warning[model]['replacement'])
```
This will include extra whitespace at the end of the string.

clinssen commented on pynest/nest/lib/hl_api_info.py:
```diff
@@ -89,9 +90,11 @@ def helpdesk():
+        text = textwrap.dedent("""\
+            NEST help needs to know where NEST is installed.
+            Please source nest_vars.sh or define NEST_DOC_DIR manually
+        """)
```
unnecessary use of dedent()

stinebuu replied:
@clinssen If I do not use `dedent()` the indenting of the printed text is a bit strange. Do you have any tips on how to get the best results for the printed text? The best I have managed is the one below. As you can see, there are empty spaces before the text starts, but at least the two sentences are on the same line. (This is without dedent())

I have the same problem with indenting for the print below as well.

![image](https://user-images.githubusercontent.com/21308850/84747451-8c8abd00-afb7-11ea-8d84-67aa43a112e6.png)

clinssen replied:
Everything between the triple quotes will become part of the string, including indentation and newlines, so if we use triple-quoted strings, we should use 
```Python
        text = """\
NEST help needs to know where NEST is installed.
Please source nest_vars.sh or define NEST_DOC_DIR manually"""
```
instead of
```Python
        text = """\
            NEST help needs to know where NEST is installed.
            Please source nest_vars.sh or define NEST_DOC_DIR manually
        """
```
I find this a bit messy and would suggest to stick with single-quoted strings, as the strings in question aren't that long.

clinssen commented on pynest/nest/lib/hl_api_info.py:
```diff
@@ -139,19 +142,18 @@ def help(obj=None, pager=None, return_text=False):
+            Type 'nest.version()' for information about the NEST version.
+
+            For more information visit https://www.nest-simulator.org.
+        """)
```
unnecessary use of dedent and temporary variable

clinssen commented on pynest/nest/lib/hl_api_connection_helpers.py:
```diff
@@ -82,58 +82,44 @@ def _process_syn_spec(syn_spec, conn_spec, prelength, postlength, connect_np_arr
-                                "dimension " + str(conn_spec['N']) + ", a "
-                                "scalar or a dictionary.")
+                                "'" + key + "' has to be an array of dimension " + str(conn_spec['N']) +
+                                ", a scalar or a dictionary.")
```
Not a fan of this option, as then the code formatting becomes rather ugly (string indentation does not match code indentation), or you end up making unnecessary calls to textwrap.dedent() (see below).

No opinion on using str.format() vs. string concatenation.

stinebuu commented:
@hakonsbm, @clinssen Thanks for your comments! I have tried to switch to single-quoted strings and use format. Please take another look! There might be things I have overlooked.

hakonsbm commented on pynest/nest/lib/hl_api_helper.py:
```diff
@@ -515,10 +514,9 @@ def model_deprecation_warning(model):
-            future version of NEST, use {1} instead.\
-            ".format(model, _deprecation_warning[model]['replacement'])
-            text = get_wrapped_text(text)
+            text = \
+                "The {0} model is deprecated and will be removed in a future version of NEST, use {1} instead.".format(
+                    model, _deprecation_warning[model]['replacement'])
```
I'm not sure that breaking the line with `\` here is the right way to go, I think the preferred way is to use parentheses.

clinssen replied:
Parentheses can be avoided:
```Python
            text = "The {0} model is deprecated and will be removed in a future version of NEST, " \
                   "use {1} instead.".format(model, _deprecation_warning[model]['replacement'])
```

hakonsbm replied:
We want to avoid the backslash. From [PEP8:](https://www.python.org/dev/peps/pep-0008)

> The preferred way of wrapping long lines is by using Python's implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation.

And it looks like `.format()` works with implicitly continued strings.

clinssen replied:
Personally I prefer the option that avoids brackets, but PEP8 is of course the standard for NEST Simulator, so I am dropping this point! Tabs versus spaces, however... :)

clinssen commented on pynest/nest/lib/hl_api_helper.py:
```diff
@@ -117,9 +117,8 @@ def show_deprecation_warning(func_name, alt_func_name=None, text=None):
-                text = "{0} is deprecated and will be removed in a future \
-                version of NEST.\nPlease use {1} instead!\
-                ".format(func_name, alt_func_name)
+                text = """{0} is deprecated and will be removed in a future version of NEST.
```
Please replace triple-quoted string by single-quoted string(s).

stinebuu commented:
@hakonsbm @clinssen I have adressed your comments, please have another look.

There are a couple of places I have replaced `'` with `"` *or* `"` with `'`. I know this looks strange, but it is for consistency *within* the function. If I have changed the quote it is because other strings in the function use the other type. Let me know if you want me to change it back. :+1: 

Also, Travis is complaining about a bare except in `hl_api_nodes`, ` pynest/nest/lib/hl_api_nodes.py:120:9: E722 do not use bare except'`. I have not changed this except, but I have changed the file. Changing the `except` is not really part of this PR, but it is making the test fail. Any thoughts of what is best to do here?

clinssen commented:
Please clean up the bare except in a "trivial" way for now by replacing ``try..except:`` by ``try..except Exception:``.

stinebuu commented:
@clinssen This was solved in #1653, which is now merged. I have merged master, so it should be all good now :+1:
